### PR TITLE
Feature integration data table

### PR DESCRIPTION
**Pull request by ketankartoza**
This PR is for #573

Added API on Backend
Added UI and API Integration on Frontend
User also able to download dataTable in both formate csv and xlsx 

Before : 
Screenshot 2023-07-12 at 1 31 52 PM

After:
Screenshot 2023-07-12 at 12 56 27 PM

This SS for Showing selected columns on data table 
Screenshot 2023-07-12 at 1 36 01 PM

### CODE REVIEW

**Comment by ketankartoza** [comment]
> Hi @ketankartoza, could you also share a GIF or screencast? This is a pretty large PR, so it would be better to see it in action.

Hi @dimasciput, here is the recording. Please check.
https://github.com/kartoza/sawps/assets/138113337/29ae5112-1bd1-4111-ad81-f7af68ce7656

File: django_project/frontend/src/containers/MainPage/Data/index.tsx [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+            offtake_permit_number: '',
+            notes: '',
+        };
+    });
+
+    const fetchDataList = () => {
+        setLoading(true)
+        axios.get(`${FETCH_AVAILABLE_DATA}?month=${selectedMonths}&start_year=${startYear}&end_year=${endYear}&species=${speciesParam}`).then((response) => {
+            setLoading(false)
+            if (response.data) {
```
Comment (dimasciput) [inline django_project/frontend/src/containers/MainPage/Data/index.tsx:151]: If there's no data, the data should be empty, not retain the old one.

Comment (ketankartoza) [reply django_project/frontend/src/containers/MainPage/Data/index.tsx:151]: @dimasciput I checked and i think there is an empty table retains when there is no data available

File: django_project/frontend/api_views/data_table.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+from species.models import OwnedSpecies
+from frontend.serializers.data_table import OwnedSpeciesSerializer
+from frontend.filters.data_table import OwnedSpeciesFilter
+
+
+class DataTableAPIView(APIView):
+    permission_classes = [IsAuthenticated]
+    filter_class = OwnedSpeciesFilter
+    queryset = OwnedSpecies.objects.all()
```
[the remark this answers is not in the document]
Comment (ketankartoza) [reply django_project/frontend/api_views/data_table.py:13]: Okay

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 18 min

File: django_project/frontend/api_views/data_table.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+class DataTableAPIView(APIView):
+    permission_classes = [IsAuthenticated]
+    filter_class = OwnedSpeciesFilter
+
+    def get_queryset(self):
+        organisation_id = self.request.session.get('current_organisation_id')
+        queryset = OwnedSpecies.objects.select_related('user').filter(
+            user__organisationuser__organisation_id=organisation_id
```
Comment (danangmassandy) [inline django_project/frontend/api_views/data_table.py:16]: @ketankartoza query would be the OwnedSpecies that are in property belongs to current organisation. That way you can get all properties inside organisation.

**Review by danangmassandy — CHANGES_REQUESTED** [review]

File: django_project/frontend/api_views/data_table.py (hunk 3 above)
Comment (ketankartoza) [reply django_project/frontend/api_views/data_table.py:16]: Like this?
queryset = OwnedSpecies.objects.filter(property__organisation_id=organisation_id

Comment (danangmassandy) [reply django_project/frontend/api_views/data_table.py:16]: yes, that would be correct @ketankartoza

Comment (ketankartoza) [reply django_project/frontend/api_views/data_table.py:16]: okay

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 12 min

File: django_project/frontend/api_views/data_table.py (hunk 3 above)
Comment (danangmassandy) [reply django_project/frontend/api_views/data_table.py]: @ketankartoza the queryset is still not changed to use organisation from property

**Review by danangmassandy — CHANGES_REQUESTED** [review]

**Review by danangmassandy — APPROVED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 16 min

**Pull request merged**